UsbIpServer/Program.cs

```diff
@@ -117,97 +121,149 @@ void DefaultCmdLine(CommandLineApplication cmd)
                 cmd.Description = "Bind device";
                 var busId = cmd.Option("-b|--busid=<busid>", "Share device having <busid>", CommandOptionType.SingleValue);
                 var bindAll = cmd.Option("-a|--all", "Share all devices.", CommandOptionType.NoValue);
+                var temporary = cmd.Option("-t|--temporary", "Do not automatically share this device if it is reconnected.", CommandOptionType.NoValue);
                 DefaultCmdLine(cmd);
-                cmd.OnExecute(async () =>
+                cmd.OnExecute(() => BindDeviceAsync(bindAll.HasValue(), busId.Value(), temporary.HasValue(), CancellationToken.None));
+            });
+
+            app.Command("unbind", (cmd) =>
+            {
+                cmd.Description = "Unbind device";
+                var busId = cmd.Option("-b|--busid=<busid>", "Stop sharing device having <busid>", CommandOptionType.SingleValue);
+                var guid = cmd.Option("-g|--guid=<guid>", "Stop sharing persisted device having <guid>", CommandOptionType.SingleValue);
+                var unbindAll = cmd.Option("-a|--all", "Stop sharing all devices.", CommandOptionType.NoValue);
+                DefaultCmdLine(cmd);
+                cmd.OnExecute(() => UnbindDeviceAsync(unbindAll.HasValue(), busId.HasValue() ? busId.Value() : null, guid.HasValue() ? guid.Value() : null, CancellationToken.None));
+            });
+
+            app.Command("server", (cmd) =>
+            {
+                cmd.Description = "Run the server stand-alone on the console";
+                DefaultCmdLine(cmd);
+                cmd.Argument("key=value", ".NET configuration override", true);
+                cmd.OnExecute(() => ExecuteServer(cmd.Arguments.Single().Values.ToArray()));
+            });
+
+            app.Command("wsl", (metacmd) =>
```

Comment:
I added some basic sanity checks for all the `wsl` commands that check if `wsl.exe` exists where we expect it to. If it does not, then a message is shown with a link to WSL installation instructions.

This doesn't do any special handling for the case where only WSL 1 is available, but if you're okay with it I'm comfortable merging this as-is now and I can follow up with the WSL PM about how we want to handle error messaging for different versions of WSL.

Reply:
Okay, I'm doing a little more digging and I'm going to see if it works to use [WslGetDistributionConfiguration](https://docs.microsoft.com/en-us/windows/win32/api/wslapi/nf-wslapi-wslgetdistributionconfiguration) to automatically detect the distro version. Unfortunately the API is fairly weak and we can't use it for other things like [listing the available distros](https://github.com/microsoft/WSL/issues/4644), but in this case it might work.

Reply:
Updated to detect WSL 1 distros during the attach command and show an appropriate error message.